src/components/Picklist/__test__/picklist.spec.js

```diff
@@ -48,7 +58,497 @@ describe('<Picklist />', () => {
+            },
+        );
+        component.instance().openMenu();
+        component.update();
+
+        expect(component.state().isOpen).toBe(true);
+        expect(component.find("div[role='presentation']").prop('className')).toBe(
+            'rainbow-picklist rainbow-picklist--open',
+        );
+
+        component.find('input').simulate('keyDown', {
+            keyCode: ESCAPE_KEY,
+        });
+
+        expect(component.state().isOpen).toBe(false);
+        expect(component.find("div[role='presentation']").prop('className')).toBe(
+            'rainbow-picklist',
+        );
+    });
+    it('should close menu when is opened and press TAB key', () => {
+        const component = mount(
+            <Picklist label="Picklist">
+                <PicklistOption label="Option 1" name="option1" />
+                <PicklistOption label="Option 2" name="option2" />
+                <PicklistOption label="Option 3" name="option3" />
+            </Picklist>,
+        );
+        jest.runAllTimers();
+
+        component.instance().menuRef.current = Object.assign(
+            {},
+            component.instance().menuRef.current,
+            {
+                clientHeight: 225,
+                scrollTop: 0,
+                scrollHeight: 285,
+                scrollTo: jest.fn(() => {}),
+            },
+        );
+        component.instance().openMenu();
+        component.update();
+
+        expect(component.state().isOpen).toBe(true);
+        expect(component.find("div[role='presentation']").prop('className')).toBe(
+            'rainbow-picklist rainbow-picklist--open',
+        );
+
+        component.find('input').simulate('keyDown', {
+            keyCode: TAB_KEY,
+        });
+
+        expect(component.state().isOpen).toBe(false);
+        expect(component.find("div[role='presentation']").prop('className')).toBe(
+            'rainbow-picklist',
+        );
+    });
+    it('should fire onChange when option is selected by click', () => {
+        const onChangeFn = jest.fn();
+        const component = mount(
+            <Picklist label="Picklist" onChange={onChangeFn}>
+                <PicklistOption label="Option 1" name="option1" />
+                <PicklistOption label="Option 2" name="option2" />
+                <PicklistOption label="Option 3" name="option3" />
+            </Picklist>,
+        );
+        jest.runAllTimers();
+        component.instance().menuRef.current = Object.assign(
+            {},
+            component.instance().menuRef.current,
+            {
+                clientHeight: 225,
+                scrollTop: 0,
+                scrollHeight: 285,
+                scrollTo: jest.fn(() => {}),
+            },
+        );
+        component.instance().openMenu();
+        component.update();
+        component
+            .find('li.rainbow-picklist-option')
+            .at(1)
+            .simulate('mousedown');
+        expect(onChangeFn).toHaveBeenCalled();
```

Comment:
better use check what data is send in onChange and change the name to:
```should fire onChange with the right data when option is selected by click```
and the expect:
```expect(onChangeFn).toHaveBeenCalledWith({ ... });```